subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/resolution_rules.adoc

```diff
@@ -183,6 +183,62 @@ include::sample[dir="snippets/dependencyManagement/customizingResolution-conditi
 Note that a project that is substituted must be included in the multi-project build (via `settings.gradle`).
 Dependency substitution rules take care of replacing the module dependency with the project dependency, but do not implicitly include the project in the build.
 
+[[sec:substitution_with_classifier]]
+=== Substituting a dependency with a classifier or artifact
+
+While external modules are in general addressed via their group/artifact/version coordinates, it is common that such modules are published with additional artifacts that you may want to use in place of the main artifact.
+This is typically the case for _classified_ artifacts, but you may also need to select an artifact with a different file type or extension.
```

Comment:
I wonder myself what the practical differente between _file type_ and _extension_ is?

Reply:
I'm not quite sure either, it seems to be mostly relevant for Ivy.

Reply:
In Ivy, it allows you to do something like `type=docs`, `ext=jar` for the sources and javadoc jars for example. These artifacts could then be located in a different folder on the server. Of course not really with Gradle since we do not fully support `type` 😉